Sources/NIO/Selector.swift

```diff
@@ -41,6 +41,71 @@ private extension Optional {
     }
 }
 
+private struct KQueueEventFilterSet: OptionSet, Equatable {
+    typealias RawValue = UInt8
+
+    let rawValue: UInt8
+
+    static let _none = KQueueEventFilterSet(rawValue: 0)
+    static let except = KQueueEventFilterSet(rawValue: 1 << 1)
```

Comment:
Why does this shift start from 1?

Reply:
because there's a loose correlation between `SelectorEventSet`, `KQueueEventFilterSet` and `KQueueEventFilterSet`:

- `hangup` is `hangup` for epoll and `except` for kqueue
- `readEOF` is `readHangup` for epoll and _also_ `except` for kqueue (but `except` most closely corresponds to `readHangup`)
- `read` is `input` for epoll and `read` for kqueue
- `write` is `output` for epoll and `write` for kqueue

So for kqueue you only need/have three filters but we and epoll represent four different ones. I chose the same bit values for the things that correspond most closely so that debugging is slightly easier.

Writing this made me realise a bug: I didn't set `except` if we _only_ waited for `reset` and I should. Making a test

Reply:
So this reasoning is fine, but can you leave a comment? I'll definitely forget this in a month.